### PR TITLE
Octavia provider driver capability to provision F5 BigIP Layer2 entities

### PR DESCRIPTION
**Pull request by notandy**
This PR introduces a new network driver that is able to provision and de-provision l2 entities like

* VLANs
* Route-Domains
* Routes
* SelfIPs

and supports VCMP Hosts and Guest. It uses taskflow to provision the devices and is able to take over the work of networking-f5 neutron plugin. 

By letting octavia-worker do the provisioning, we eliminate the asynchronous behaviour of the networking-f5 and octavia-worker actions and ensuring that AS3 declarations are submitted with required route-domains / l2 configuration.

Also, migration of loadbalancer between devices should be easier and faster to implement.

### CODE REVIEW

File: octavia_f5/network/drivers/neutron/neutron_client.py [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+                             load_balancer.vip.port_id)
+                    return self._port_to_vip(port, load_balancer)
+                LOG.error('Neutron VIP mis-match. Expected ip %s on '
+                          'subnet %s in network %s. Neutron has fixed_ips %s '
+                          'in network %s. Deleting and recreating the VIP '
+                          'port.', load_balancer.vip.ip_address,
+                          load_balancer.vip.subnet_id,
+                          load_balancer.vip.network_id,
+                          self._fixed_ips_to_list_of_dicts(port.fixed_ips),
+                          port.network_id)
```
Comment (BenjaminLudwigSAP) [inline octavia_f5/network/drivers/neutron/neutron_client.py:113]: Does this really need to be in the try-block?

File: octavia_f5/network/drivers/neutron/hierarchical_port_binding.py [hunk 2]
```diff
@@ -1,173 +0,0 @@
-# Copyright 2018 SAP SE
```
Comment (BenjaminLudwigSAP) [inline octavia_f5/network/drivers/neutron/hierarchical_port_binding.py:1]: Please remove hierarchical_port_binding.py completely, you just emptied it

File: octavia_f5/network/drivers/neutron/neutron_client.py (hunk 1 above)
Comment (notandy) [reply octavia_f5/network/drivers/neutron/neutron_client.py:113]: yes - to comply with the upstream implementation. It's mostly c&p from the allowed_address_pairs driver: https://github.com/sapcc/octavia/blob/stable/ussuri-m3/octavia/network/drivers/neutron/allowed_address_pairs.py#L455

File: octavia_f5/network/drivers/neutron/hierarchical_port_binding.py (hunk 2 above)
Comment (notandy) [reply octavia_f5/network/drivers/neutron/hierarchical_port_binding.py:1]: good catch

**Pull request merged**